### PR TITLE
Add a recommendation for the ICU version

### PR DESCRIPTION
**Pull request by jakzal**
Fixes https://github.com/symfony/symfony-standard/issues/952

### CODE REVIEW

File: Resources/skeleton/app/SymfonyRequirements.php [hunk 1]
```diff
@@ -677,6 +677,14 @@ function_exists('posix_isatty'),
                 'Upgrade your <strong>intl</strong> extension with a newer ICU version (4+).'
             );
 
+            if (class_exists('Symfony\Component\Intl\Intl')) {
+                $this->addRecommendation(
+                    \Symfony\Component\Intl\Intl::getIcuDataVersion() === \Symfony\Component\Intl\Intl::getIcuVersion(),
+                    sprintf('intl ICU version installed on your system (%s) should match the ICU data bundled with Symfony (%s)', \Symfony\Component\Intl\Intl::getIcuVersion(), \Symfony\Component\Intl\Intl::getIcuDataVersion()),
+                    sprintf('Upgrade your <strong>intl</strong> extension with a matching ICU version (%s).', \Symfony\Component\Intl\Intl::getIcuDataVersion())
```
Comment (fabpot) [inline Resources/skeleton/app/SymfonyRequirements.php]: Is it actionnable? I mean, is it easy to upgrade ICU to a newer/older version than the one shipped by default on Linux/Mac?

Comment (jakzal) [inline Resources/skeleton/app/SymfonyRequirements.php]: Actually, it's not :( you'd need to compile it (icu and php).

Perhaps we should just warn that some data might be inconsistent and needs to be verified for locale used in a project?

Comment (fabpot) [inline Resources/skeleton/app/SymfonyRequirements.php]: Indeed, I would then just warn the user that he might have some inconsistencies.

**Comment by fabpot** [comment]
Thank you @jakzal.

**Comment by shawe** [comment]
@fabpot @jakzal the warning is added, but... how we solve it?

**Comment by julinoneto** [comment]
Like @shawe said: the warning is added, but... how we solve it?

**Comment by jakzal** [comment]
@shawe @julinoneto as the message says you need to ipgrade your intl extension with a version that matches Symfony's ICU version (55.1 at the moment). Please suggest an improved message if you find it unclear.

**Comment by julinoneto** [comment]
@jakzal i am trying to do it on a CentOS 7 O.S. but i cant find a way to upgrade my ICU version on the O.S. using PHP 7.

**Comment by shawe** [comment]
I'm using ArchLinux, always with latest versions of software. For me this problem seems to not be logical.

**Comment by jakzal** [comment]
@shawe this is not necessarily a problem. It's just a suggestion. 

It's best to be on the latest version of ICU as it's the only one that's up to date. Not all ICU data is exposed by the intl extension and Symfony provides ways to access it. This means that we need to bundle that data with the component and it might be that your system has a different version installed.

The warning generated by Symfony is just to point out there MIGHT be slight inconsistencies for rather rare edge cases.

**Pull request merged**